### PR TITLE
Bugfix/join on source relation

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:** 
- #94 

**This PR will result in the following new package version:**

- 11.0 - Could use input on this. On the one hand no changes for most users, but on the other hand this could affect row counts for users using union feature. I chose to err on the side of caution but open to suggestions.

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**
- Adds `source_relation` to appropriate joins, window function partitions, and window function order bys. 

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [ ] ~dbt compile~ Does not compile, known issue: #83 
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [x] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.

- See workbook linked in ticket.

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [x] README updates have been applied (if applicable)
    
- [x] DECISIONLOG updates have been updated (if applicable)
- [x] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [x] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: models/quickbooks__cash_flow_statement.sql [hunk 1]
```diff
@@ -7,8 +7,10 @@ with cash_flow_classifications as (
 final as (
     
     select cash_flow_classifications.*,
-        coalesce(lag(cash_ending_period) over (partition by account_id, class_id, source_relation order by cash_flow_period),0) as cash_beginning_period,
-        cash_ending_period - coalesce(lag(cash_ending_period) over (partition by account_id, class_id, source_relation order by cash_flow_period), 0) as cash_net_period
+        coalesce(lag(cash_ending_period) over (partition by account_id, class_id, source_relation 
+            order by source_relation, cash_flow_period),0) as cash_beginning_period,
```
Comment (fivetran-avinash) [inline models/quickbooks__cash_flow_statement.sql:11]: good catch to order it by source relation, missed this when I did it.

File: models/quickbooks.yml [hunk 2]
```diff
@@ -283,14 +289,16 @@ models:
...
     description: "{{ doc('general_ledger_table') }}"
-    tests:
-      - dbt_utils.unique_combination_of_columns:
-          combination_of_columns:
-            - unique_id
-            - source_relation   
     columns:
       - name: unique_id
-        description: Unique identifier of the general ledger line, dependent on `transaction_id`, `transaction_index`, `account_id`, `transaction_type`, `transaction_source`. 
+        description: >
```
Comment (fivetran-avinash) [inline models/quickbooks.yml:294]: Good call that the above test is redundant.

File: models/quickbooks__cash_flow_statement.sql (hunk 1 above)
Comment (fivetran-avinash) [inline models/quickbooks__cash_flow_statement.sql:11]: good catch to order it by source relation, missed this when I did it.

**Review by fivetran-avinash — APPROVED** [review]
I think this is mostly good to go! Just need to update the CHANGELOG  --it is definitely a breaking change, good intuition!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,18 @@
+# dbt_quickbooks v0.11.0
+[PR #95](https://github.com/fivetran/dbt_quickbooks/pull/95) includes the following updates:
+## 🎉 Features
+- Added description for column `source_relation` to the documentation.
+
+## 🪲 Bug Fixes
+- Included `source_relation` in all joins and window functions for models outputting `source_relation`. This is to prevent duplicate records in end models when using the unioning functionality. These updates were in the intermediate models, which flowed to downstream end models:
+  - `quickbooks__general_ledger`
+  - `quickbooks__expenses_sales_enhanced`
+- In end model `quickbooks__general_ledger`, added `source_relation` as part of the generated surrogate key `unique_id` to prevent duplicate `unique_id`s when using the unioning functionality.
```
[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply CHANGELOG.md:10]: Agreed. Updated!

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,18 @@
...
+
+## 🚘 Under the Hood
+- Updated test from a combination of columns to uniqueness of `unique_id` in `quickbooks__general_ledger`.
```
[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply CHANGELOG.md:13]: Good call. Updated.

**Pull request merged**